File: ruby/lib/BetaPowerEfficientSpaceStation.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'Dice'
require_relative 'PowerEfficientSpaceStation'
require_relative 'PowerEfficientSpaceStationToUI'

module Deepspace
class BetaPowerEfficientSpaceStation < PowerEfficientSpaceStation
  @@EXTRAEFFICIENCY = 1.2
  
  def initialize(station)
    super
    @dice = Dice.new()
  end
  
  def fire
    shot = super
    
    if @dice.extraEfficiency
      shot *= @@EXTRAEFFICIENCY
    end
    
    shot
      
  end
  
  def getUIversion
    BetaPowerEfficientSpaceStationToUI.new(self)
  end
  
end
end

File: java/src/deepspace/BetaPowerEfficientSpaceStation.java
/*
 * Here comes the text of your license
 * Each line should be prefixed with  * 
 */
package deepspace;

/**
 *
 * @author yabir
 */
public class BetaPowerEfficientSpaceStation extends PowerEfficientSpaceStation{

    private static final float EXTRAEFFICIENCY = 1.2f;
    Dice dice;
    
    BetaPowerEfficientSpaceStation(SpaceStation s){
        super(s);
        dice = new Dice();
    }
    
    @Override
    public float fire(){
        float factor = 1.0f;
        if(dice.extraEfficiency())
            factor = EXTRAEFFICIENCY;
        
        return super.fire()*factor;
    }
    
    @Override
    public BetaPowerEfficientSpaceStationToUI getUIversion(){
        return new BetaPowerEfficientSpaceStationToUI(this);
    }
    
    
    
}
